FHIR/src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Visitors/AnonymizationVisitor.cs

```diff
@@ -85,34 +89,109 @@ private ProcessResult ProcessResourceNode(ElementNode node)
                     continue;
                 }
 
-                IEnumerable<ElementNode> matchNodes;
-                if (rule.IsResourceTypeRule)
+                var matchNodes = GetMatchNodes(rule, node);
+                foreach (var matchNode in matchNodes)
                 {
-                    /** 
-                     * Special case handling:
-                     * Senario: FHIR path only contains resourceType: Patient, Resource. 
-                     * Sample AnonymizationFhirPathRule: { "path": "Patient", "method": "keep" }
-                     * 
-                     * Current FHIR path lib do not support navigate such ResourceType FHIR path from resource in bundle.
-                     * Example: navigate with FHIR path "Patient" from "Bundle.entry[0].resource[0]" is not support
-                     */
-                    matchNodes = new List<ElementNode>() { node };
+                    resultOnRule.Update(ProcessNodeRecursive((ElementNode)matchNode.ToElement(), _processors[method], context, rule.RuleSettings));
```

Comment:
Is it possible the matchNode be a scopedNode?

Reply:
Yes, it is converted to ElementNode with `ToElement()`.